src/aap_eda/api/serializers/eda_credential.py

```diff
@@ -20,11 +21,32 @@
 from aap_eda.core.utils.crypto.base import SecretValue
 
 
+class EdaCredentialResourceSerializer(serializers.Serializer):
+    type = serializers.CharField(
+        required=True, help_text="Type of the related resource"
+    )
+    id = serializers.IntegerField(
+        required=True, help_text="ID of the related resource"
+    )
+    name = serializers.CharField(
+        required=True, help_text="Name of the related resource"
+    )
+    url = serializers.URLField(
+        required=True, help_text="URL of the related resource"
+    )
+
+
+@extend_schema_field(EdaCredentialResourceSerializer(many=True))
+class EdaCredentialResourceField(serializers.JSONField):
+    pass
+
+
 class EdaCredentialSerializer(serializers.ModelSerializer):
     inputs = serializers.SerializerMethodField()
     credential_type = CredentialTypeRefSerializer(
         required=False, allow_null=True
     )
+    references = EdaCredentialResourceField(required=False)
```

Comment:
@hsong-rh Should this be a list?

Reply:
no, it's defined as many=True at 
https://github.com/ansible/eda-server/pull/781/files#diff-6d046528204cfa0636e34d965855b6bf4244b4c4a361ac575a1122936d76bac4R39